### PR TITLE
Added a group filter to the to do section and assignment overview

### LINKED ISSUE #934: A "nothing todo" display when all todo's are done
**Describe the solution you'd like**
A "nothing todo" display when all todo's are done. Even better would be to also add a section of "global" stats: stats about entries not graded in your group specifically 

Something like this in the ToDo section if everything is graded / filled in / whatever other todo stuff there is
image

### PR DESCRIPTION
**Pull request by engelhamer**
## Description
Added the ability to toggle filtering on a teacher's own course groups to the to do section and assignment overview. This allows teachers to view what is to be done for the groups they are responsible for at a glance, instead of having to first visit the assignment page and then filtering on their own groups there.

Also added a message when all to do's are done.

image

image

## Summary of changes
- Fixes #934
- Added persistent filter in local storage

### CODE REVIEW

**Comment by engelhamer** [comment]
Tests will need to be added still (for the updated serializer data)

File: src/vue/src/components/assets/DeadlineDeck.vue [hunk 1]
```diff
@@ -1,40 +1,80 @@
...
+                <icon
+                    name="check"
+                    scale="4"
+                    class="fill-green mb-2 text-shadow"
+                /><br/>
+                <b class="field-heading">
+                    All done!
+                </b>
+            </div>
+            You do not have any {{ isTeacher ? 'entries to grade' : 'upcoming deadlines' }} at this moment.
```
Comment (Larspolo) [inline src/vue/src/components/assets/DeadlineDeck.vue:76]: preferably also an indicator that it is only for the group that are being filtered, and maybe even if there are more in other groups, also state that other groups sitll have some stuff to do

File: src/vue/src/components/assets/DeadlineDeck.vue [hunk 2]
```diff
@@ -93,7 +156,7 @@ export default {
 
             return deadlines
         },
-        showSortBy () {
+        isTeacher () {
             return Object.entries(this.$store.getters['user/permissions']).some(
                 ([key, value]) => ((key.indexOf('assignment') >= 0) && value.can_grade))
```
Comment (Larspolo) [inline src/vue/src/components/assets/DeadlineDeck.vue:161]: What is this index of check with checking if someone is a teacher? Either a comment, or a change in function name would be preferred

File: src/vue/src/views/AssignmentsOverview.vue [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+                <b>own groups</b>
+            </span>
+            <span
+                v-else
+                v-b-tooltip.hover
+                title="Showing to do items for all groups"
+                @click="selectedFilterOwnGroups = true"
+            >
+                Showing to do:
+                <b>all</b>
```
Comment (Larspolo) [inline src/vue/src/views/AssignmentsOverview.vue:23]: This is now duplicate code of elsewhere right? If so, would prefer their own component to reduce code clones.

File: src/django/VLE/serializers.py [hunk 4]
```diff
@@ -363,21 +363,23 @@ def get_stats(self, assignment):
...
-        if participation.exists():
-            participation = participation.first()
-            if participation.groups and users.filter(participation__groups=participation.groups.first()).exists():
-                users = users.filter(participation__groups=participation.groups.first())
+        own_participation = Participation.objects.filter(user=self.context['user'], course=course)
+        if own_participation.exists():
+            # Get all users that are in any of the request user's groups.
+            own_participation = own_participation.first()
+            own_group_users = all_users.filter(participation__groups__in=own_participation.groups.all())
+            print('\n\n\n', own_group_users.values(), own_participation.groups.all())
```
Comment (Larspolo) [inline src/django/VLE/serializers.py:379]: :(

Comment (engelhamer) [reply src/django/VLE/serializers.py:379]: Whoops forgot to push removal commit

File: src/vue/src/views/AssignmentsOverview.vue (hunk 3 above)
Comment (engelhamer) [reply src/vue/src/views/AssignmentsOverview.vue:23]: This is merely a very specific clickable text so I do not really think that we would benefit much from making it into a separate component.

File: src/django/VLE/serializers.py [hunk 5]
```diff
@@ -363,21 +363,22 @@ def get_stats(self, assignment):
 
         course = self.context.get('course', None)
         if course is None:
+            # Get the stats from only the course that its linked to, when no courses are supplied.
```
Comment (mjvkeulen) [inline src/django/VLE/serializers.py:366]: it's

File: src/vue/src/components/assets/DeadlineDeck.vue [hunk 6]
```diff
@@ -1,40 +1,83 @@
...
+            v-if="$root.canGradeForSomeCourse"
+            class="text-grey float-right unselectable cursor-pointer"
         >
-            <b-link
-                :to="$root.assignmentRoute(d)"
-                tag="b-button"
+            <span
+                v-if="filterOwnGroups"
+                v-b-tooltip.hover
+                title="Showing to do items only for groups which you are a member of"
```
Comment (mjvkeulen) [inline src/vue/src/components/assets/DeadlineDeck.vue:16]: Drop only mid-sentence or move to front: `Only showing to do items for groups which you are a member of`

File: src/vue/src/components/assets/TodoCard.vue [hunk 7]
```diff
@@ -22,7 +22,9 @@
         <br/>
         <span v-if="deadline.deadline.date">
             <!-- Teacher deadline shows last submitted entry date  -->
-            <span v-if="deadline.stats.needs_marking + deadline.stats.unpublished > 0">
+            <span
+                v-if="unpublishedOrNeedsMarking"
```
Comment (mjvkeulen) [inline src/vue/src/components/assets/TodoCard.vue:26]: Should single property tags be forced on a single line?

File: src/vue/src/components/assets/DeadlineDeck.vue (hunk 6 above)
Comment (engelhamer) [reply src/vue/src/components/assets/DeadlineDeck.vue:16]: Matter of personal preference

**Pull request merged**